readthedocs/settings/base.py

```diff
@@ -500,12 +500,17 @@ def TEMPLATES(self):
         },
     }
 
-    # Convert all `python.supported_versions` to string for backward compatiblity
-    # see https://github.com/readthedocs/readthedocs.org/pull/8328#discussion_r666366384
-    for k in DOCKER_IMAGE_SETTINGS:
-        for version in DOCKER_IMAGE_SETTINGS[k]['python']['supported_versions']:
-            if isinstance(version, (int, float)):
-                DOCKER_IMAGE_SETTINGS[k]['python']['supported_versions'].append(str(version))
+    # Convert all `python.supported_versions` to string for backward compatiblity with numeric versions,
+    # see https://github.com/readthedocs/readthedocs.org/pull/8328#discussion_r666366384.
```

Comment:
I think we should just change these directly, because they won't change the values if we are overriding them from another file anyway.

Reply:
I don't fully understand this comment, can you explain a bit more? (What is `these` and `they`)

Reply:
The values from `DOCKER_IMAGE_SETTINGS` are only changed on this file (from number to str), so I think makes sense to just modify those values where we are declaring `DOCKER_IMAGE_SETTINGS` instead of changing them here.

Reply:
Yeah, we could duplicate all the values in the list: str and float. With the new approach, this chunk of code is not needed anymore.

It would be like:

```python
'supported_versions': [
    2,   2.7,   3,   3.5,   3.6,   3.7,
   '2', '2.7', '3', '3.5', '3.6', '3.7',
],
'default_version': {
  '2': '2.7',
  '3': '3.7',
},
```

for all the images.

Reply:
Note that I changed the `default_version` as well. We shouldn't use int/float on those either.